.github/workflows/publish.yaml

```diff
@@ -1,17 +1,41 @@
-name: 🚀 Publish
+name: 🚢 Publish
 
 on:
   push:
     branches:
       - main
-    tags:
-      - '**'
 
 jobs:
+  hasTag:
+    name: 🏷️ Analyze Tags
+    runs-on: ubuntu-24.04
+
+    outputs:
+      hasTag: ${{ steps.checkTagExists.outputs.has-tag }}
+
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v4
+        with:
+          fetch-tags: true
+
+      # ⚠️ This check works only if the tag is pushed alongside the commit
+      # Please, use `git push && git push --tags` or `git push --follow-tags` (works only for annonated tags)
+      - name: Check Tag On Commit
+        id: checkTagExists
+        run: |
+          if [ -n "$(git tag --points-at HEAD)" ]; then
+            echo "has-tag=true" >> $GITHUB_OUTPUT
```

Comment:
is it ok that this is `has-tag` and below is `hasTag`? 🤷

Reply:
`need.<job-name>.outputs.<outputs-name>`

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/passing-information-between-jobs